mlrun/serving/routers.py

```diff
@@ -708,28 +711,60 @@ def validate(self, request):
         return request
 
 
-def _init_endpoint_record(graph_server, voting_ensemble: VotingEnsemble):
-    logger.info("Initializing endpoint records")
+def _init_endpoint_record(
+    graph_server: GraphServer, voting_ensemble: VotingEnsemble
+) -> Union[str, None]:
+    """
+    Initialize model endpoint record and write it into the DB. In general, this method retrieve the unique model
+    endpoint ID which is generated according to the function uri and the model version. If the model endpoint is
+    already exist in the DB, we skip the creation process. Otherwise, it writes the new model endpoint record to the DB.
+
+    :param graph_server:    A GraphServer object which will be used for getting the function uri.
+    :param voting_ensemble: Voting ensemble serving class. It contains important details for the model endpoint record
+                            such as model name, model path, model version, and the ids of the children model endpoints.
+
+    :return: Model endpoint unique ID.
+    """
 
-    endpoint_uid = None
+    logger.info("Initializing endpoint records")
 
+    # Generate required values for the model endpoint record
     try:
+        # Getting project name from the function uri
         project, uri, tag, hash_key = parse_versioned_object_uri(
             graph_server.function_uri
         )
 
+        # Generating version model value based on the model name and model version
         if voting_ensemble.version:
             versioned_model_name = f"{voting_ensemble.name}:{voting_ensemble.version}"
         else:
             versioned_model_name = f"{voting_ensemble.name}:latest"
 
+        # Generating model endpoint ID based on function uri and model version
+        endpoint_uid = mlrun.utils.model_monitoring.create_model_endpoint_id(
+            function_uri=graph_server.function_uri, versioned_model=versioned_model_name
+        ).uid
+
+    except Exception as e:
+        logger.error("Failed to generate required values for model endpoint", exc=e)
+        return None
```

Comment:
where is the expected failure supposed to come from? I don't see reason why to catch all of this, and please use more specific error catching there are only two options from what I see. 

Also take a look at the original except that was used for that `try` clause, seems like you will never reach it which will cause the flow to actually raise exception instead of just logging a warning.

Reply:
Hi, the expected failure usually comes from the `parse_versioned_object_uri(graph_server.function_uri)` when you don't have a valid value for `graph_server.function_uri`. e.g. look at `test_v2_mock()`, an old system test in which we create a mock v2 server with missing details and expect to get 500 from the model response when trying to invoke it. Now, without that try and catch part, this test fails because we are applying parse_versioned_object_uri() on `graph_server.function_uri=None` (for more details: https://github.com/mlrun/mlrun/actions/runs/3224988669/jobs/5276749130 ). I can move this code to the 2nd try and catch but in that case we will have to replicate it because we need that `endpoint_uid` value in both parts. 

As for the original expect, we also log an error and return None. Note that the main difference is that in the new code we divided the original try and expect into two different parts to handle the cases in which we already have a model endpoint object. 

I'll modify the error desc but anyway let me know if something is not clear in this flow.